clang/tools/driver/cc1_main.cpp

```diff
@@ -198,13 +199,20 @@ static int PrintSupportedExtensions(std::string TargetStr) {
   std::unique_ptr<llvm::TargetMachine> TheTargetMachine(
       TheTarget->createTargetMachine(TargetStr, "", "", Options, std::nullopt));
   const llvm::Triple &MachineTriple = TheTargetMachine->getTargetTriple();
+  const llvm::MCSubtargetInfo *MCInfo = TheTargetMachine->getMCSubtargetInfo();
```

Comment:
I just thought of one more thing. What happens here if the build was configured without the target backend enabled?

For example if CMake was configred with `-DLLVM_TARGETS_TO_BUILD=ARM`, and you ask for the RISC-V extensions, does clang crash?

Though that is an unusual thing to do, but it is possible since clang knows about all possible triples even without a target backend. And besides that, clang shouldn't crash anyway. Simplest solution is to pass an empty description map I think, and add a test for that situation.

Reply:
It's possible `TheTargetMachine` is already null in this situation, and this is best fixed outside of this change.

I'll check that.

Reply:
I was wrong, clang doesn't recognise the triple if it wasn't enabled:
```
$ ./bin/clang --target=riscv64-linux-gnu --print-support
clang version 18.0.0 (https://github.com/llvm/llvm-project.git ebefe83c092e41d243829ab812bb650674e2f3d2)
Target: riscv64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/david.spickett/build-llvm-aarch64/./bin
No available targets are compatible with triple "riscv64-unknown-linux-gnu"
```

So we won't get to this code if the target backend is missing. No worries :)

Reply:
On my side on this topic I already checked this situation, yes looks protected well.

@DavidSpickett 

As sidenote here, on this topic opened by you: 

* In case of ```createTargetMachine```, notice the -mcpu="" (second argument), it will force to **return all target information**. If you specify i.e. "sandybridge" than you can only access list of features and descriptions only related to "sandybridge" subarchtecture / subset .

* I really hope LLVM will keep this (-mcpu="" => ALL) , and not cut it out (some developer will think is a "safer" API), otherwise it will be impossible to query LLVM features of **a whole target**.

LLVM looks (to humble me) very dynamic with frequent API changes, my point here is a "utilitarian view" as external user I would like to take benefits and features through API.